### DISCUSSION TITLE
Training/Validation split in minimal example

### DISCUSSION THREAD

**Discussion by nikvaessen** [Lightning-AI/pytorch-lightning#5815] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
If you still can't find what you need:     
#### What is your question?    

I think it's unclear how the training data is split into a training and validation split in the minimal example.

(https://williamfalcon.github.io/pytorch-lightning/LightningModule/RequiredTrainerInterface/#minimal-example)

Does this example use all training data for both training and validation? As far as I'm aware, this is bad practice. 

Is there some magic background process which compares the training and validation data loaders and does splitting? I skimmed through the code and couldn't find anything.

**Comment by awaelchli** [comment, marked as answer]
> Is there some magic background process which compares the training and validation data loaders and does splitting? I skimmed through the code and couldn't find anything.

No, I don't think this is happening. The dataset in the minimal example is the MNIST dataset, which only has two splits (train and test). In this example, the validation set is the same as the training set and if we wanted to split it, we would have to use something like [torch.utils.random_split](https://pytorch.org/docs/stable/data.html#torch.utils.data.random_split).

**Comment by williamFalcon** [comment]
exactly! it’s meant to show how lightning works, not how validation works :)

**Comment by nikvaessen** [comment]
Should a training/validation split be something pytorch-lightning automates for you? Has this been considered as a feature?

**Comment by AtomScott** [comment]
The [data preparation section](https://pytorch-lightning.readthedocs.io/en/latest/api/pytorch_lightning.core.html#data-preparation) (on the same page as the minimal example) describes how one would split the datasets into train/test/split.

I don't think it's something that should be automated by pytorch-lightning since you can easily implement a split in `setup()`

**Comment by noamzilo** [comment]
> 
> 
> The [data preparation section](https://pytorch-lightning.readthedocs.io/en/latest/api/pytorch_lightning.core.html#data-preparation) (on the same page as the minimal example) describes how one would split the datasets into train/test/split.
> 
> I don't think it's something that should be automated by pytorch-lightning since you can easily implement a split in `setup()`

the link does not work.

did you mean [this?](https://pytorch-lightning.readthedocs.io/en/latest/datamodules.html)
